services/brig/test/integration/API/Provider.hs

```diff
@@ -169,6 +170,58 @@ testRegisterProvider db brig = do
         assertEqual "description" defProviderDescr (providerDescr p)
         assertEqual "profile" (ProviderProfile p) pp
 
+-- | Step-by-step registration procedure with verification
+-- of pre- and post-conditions, using an internal endpoint
+-- instead of accessing the DB directly
+testRegisterProviderInternal :: Brig -> Http ()
```

Comment:
This test seems to be mostly a duplicate of `testRegisterProvider`. If you don't want to refactor them, it'd be nice to at least have a `-- TODO: duplicate of ...` comment here.